fedimintd/src/bin/main.rs

```diff
@@ -135,6 +135,7 @@ async fn main() -> anyhow::Result<()> {
 
     let decoders = module_inits.decoders();
 
+    db.register_module_decoders(decoders.clone());
```

Comment:
Since `db` isn't used before that point, why isn't `decoders` just another arg to `fedimint_rocksdb::RocksDb::open`? I really don't like to have that inconsistent DB struct that can't decode *some* of the entries.

Reply:
This is explained in the message of the third commit.
- Some modules need database during creation
- ModuleDecoderRegistry can only be created after modules have been created.

Reply:
https://github.com/fedimint/fedimint/blob/f2127a803143e3c293406f0352c93d5dfdd6f6b4/fedimint-server/src/config.rs#L178 this should help. RN they are created (db vs ModuleInitRegistry) in opposite order, but I think there's nothing preventing them from being reordered: https://github.com/fedimint/fedimint/blob/f2127a803143e3c293406f0352c93d5dfdd6f6b4/fedimintd/src/bin/main.rs#L127

Reply:
Oh, I didn't notice `decoders` on ModuleInitRegistry.